### ISSUE TITLE
Staying in sync with cakephp/app

### ISSUE THREAD

**Issue by ricog** [loadsys/CakePHP-Skeleton#87]
Just dumping my thoughts before I go eat. The last sync was at:

    cakephp/app: ea476366148e71bcfc3f346fedb450516104bb6b

So what I did is generate a patch between the last merge and now with this command run in a cakephp/app clone:

    git format-patch ea476366148e71bcfc3f346fedb450516104bb6b --stdout > app.patch

I then take this patch and apply it on our repo.

    git checkout <last merge of cakephp/app>
    git checkout -b f/app-update-74c9759615fbcfe4fea7cb4e2d4baa1815d2b75b
    git apply --stat app.patch
    git apply --check app.patch
    git am --ignore-whitespace --signoff < app.patch

This should place the commits into the new `f/app-update-74c9759615fbcfe4fea7cb4e2d4baa1815d2b75b` branch. I used this branch name to ensure the last cakephp/app hash is retained when we merge. Next it can be rebased onto our work and any conflicts can be resolved.

    git rebase origin/master

Now push and create a PR (I made #86).

These instructions should probably live in the README.md of the skeleton. Also, it might be useful to keep a reference of the last hash merged into the skeleton too. Thoughts?

**Comment by jtyost2** [comment]
Re: "These instructions should probably live in the README.md of the skeleton. Also, it might be useful to keep a reference of the last hash merged into the skeleton too. Thoughts?"

Yes and Yes.

**Comment by beporter** [comment]
+1 +1 +1, on all counts.

**Comment by beporter** [comment]
It's seriously really great to have actual concrete instructions for how to do this. And not just "go do this" but actual _commands_ to issue that do the majority of the work for you.

In fact, could this be scripted to clone/pull the cake core's skeleton, create the patch and create the local branch with the diff applied for you? All you'd have to do is rebase, resolve conflicts, and push.

**Comment by ricog** [comment]
Good idea. Do you prefer this to be in the shell script repo or here? If here, where?

**Comment by beporter** [comment]
Good question. Drat. 

This would be a script specific to working with the skeleton, and would have zero value to any other project, right? Or could it be made to provide a diff of changes from OUR skeleton compared to a "target" repo???

We might be able to get double the bang for our buck if that's possible. Maybe it'd allow us to import core changes into our skeleton, AND import changes to our skeleton into our established cake 3 apps??

That seems worth investigating, and if possible means the script belongs in the 3.0 series of the Shell Scripts (master branch now).

**Comment by ricog** [comment]
Another good idea. It could be agnostic, only needing a target repo and a few parameters. So I would have a script, maybe called `sync-upstream` or `get-upstream-changes`. Suggestions?

I would also need a config to tell the script which repo and maybe the last merged commit. I may be able to get the last merged commit programmatically.

**Comment by beporter** [comment]
I was thinking that you could check the `remote` name for the "current" repo, and if it's the skeleton, the compare against the core team's skeleton repo, otherwise compare against loadsys's skeleton repo. 

I would love it if no arguments were needed so go for programmatic as much as possible. 

As for a name, I recommend a noun-verb pair to stay consistent with the shell scripts repo. 'skeleton-diff' or something?

**Comment by ricog** [comment]
I'm throwing some manual instructions into the README. In order to make this work we will need to maintain a branch specific to the `cakephp-app` updates. Also, automating the process would involve finding the last merge commit message and parsing the <ref>. The command I use manually is:

git log -1 --oneline --grep="app-update"
